lib/game_client/ogs/ogs_game_client.dart

```diff
@@ -0,0 +1,362 @@
+        final boardSize = gameData['width'] as int? ?? 19;
+
+        // Parse dates
+        final endedStr = gameData['ended'] as String?;
+        final dateTime =
+            endedStr != null ? DateTime.parse(endedStr) : DateTime.now();
+
+        // Parse players
+        final blackPlayerData = gameData['players']['black'];
+        final whitePlayerData = gameData['players']['white'];
+
+        final blackPlayer = UserInfo(
+          userId: blackPlayerData['id'].toString(),
+          username: blackPlayerData['username'] ?? '',
+          rank:
+              _ratingToRank(blackPlayerData['ratings']?['overall']?['rating']),
+          online: false, // Not available in this API
+          winCount: 0, // Not available in this API
+          lossCount: 0, // Not available in this API
+        );
+
+        final whitePlayer = UserInfo(
+          userId: whitePlayerData['id'].toString(),
+          username: whitePlayerData['username'] ?? '',
+          rank:
+              _ratingToRank(whitePlayerData['ratings']?['overall']?['rating']),
+          online: false, // Not available in this API
+          winCount: 0, // Not available in this API
+          lossCount: 0, // Not available in this API
+        );
+
+        // Parse game result
+        final outcome = gameData['outcome'] as String? ?? '';
+        final blackLost = gameData['black_lost'] as bool? ?? false;
+        final whiteLost = gameData['white_lost'] as bool? ?? false;
+
+        wq.Color? winner;
+        if (blackLost && !whiteLost) {
+          winner = wq.Color.white;
+        } else if (whiteLost && !blackLost) {
+          winner = wq.Color.black;
+        }
+        // If both lost or neither lost, winner remains null (draw/unknown)
+
+        final result = GameResult(
+          winner: winner,
+          result: outcome,
+        );
+
+        return GameSummary(
+          id: gameId,
+          boardSize: boardSize,
+          white: whitePlayer,
+          black: blackPlayer,
+          dateTime: dateTime,
+          result: result,
+        );
+      }).toList();
+    } catch (e) {
+      throw Exception('Failed to list games: $e');
+    }
+  }
+
+  @override
+  Future<GameRecord> getGame(String gameId) async {
+    try {
+      final response = await _httpClient.get(
+        Uri.parse('$serverUrl/api/v1/games/$gameId/sgf'),
+        headers: {
+          'User-Agent': 'WeiqiHub/1.0',
+          if (_csrfToken != null) 'X-CSRFToken': _csrfToken!,
+        },
+      );
+
+      if (response.statusCode != 200) {
+        throw Exception('Failed to fetch SGF: ${response.statusCode}');
+      }
+
+      final sgfContent = response.body;
+      return _parseOgsSgf(sgfContent);
+    } catch (e) {
+      throw Exception('Failed to get game record: $e');
+    }
+  }
+
+  /// Parses OGS SGF files which structure moves as nested variations
+  /// instead of sequential nodes in the main line.
+  GameRecord _parseOgsSgf(String sgfStr) {
```

Comment:
Tbh maybe this should be fixed at the `GameRecord.fromSgf()` level.  It's not totally OGS-specific.  Any SGF with variations will fail.